modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogParamsValidationUtils.java

```diff
@@ -49,95 +44,27 @@ public static void validateIdentifier(@Nullable String identifier, String contex
-        validateZoneName(params.zoneName());
-    }
-
-    static void validateRenameZoneParams(RenameZoneParams params) {
-        validateZoneName(params.zoneName());
-        validateZoneName(params.newZoneName(), "Missing new zone name");
-    }
-
-    private static void validateUpdateZoneFieldsParameters(
-            String zoneName,
-            @Nullable Integer partitions,
-            @Nullable Integer replicas,
-            @Nullable Integer dataNodesAutoAdjust,
-            @Nullable Integer dataNodesAutoAdjustScaleUp,
-            @Nullable Integer dataNodesAutoAdjustScaleDown,
-            @Nullable String filter
-    ) {
-        validateZoneName(zoneName);
-
-        validateZonePartitions(partitions);
-        validateZoneReplicas(replicas);
-
-        validateZoneDataNodesAutoAdjust(dataNodesAutoAdjust);
-        validateZoneDataNodesAutoAdjustScaleUp(dataNodesAutoAdjustScaleUp);
-        validateZoneDataNodesAutoAdjustScaleDown(dataNodesAutoAdjustScaleDown);
-
-        validateZoneDataNodesAutoAdjustParametersCompatibility(
-                dataNodesAutoAdjust,
-                dataNodesAutoAdjustScaleUp,
-                dataNodesAutoAdjustScaleDown
-        );
-
-        validateZoneFilter(filter);
-    }
-
-    private static void validateZoneName(String zoneName) {
-        validateZoneName(zoneName, "Missing zone name");
-    }
-
-    private static void validateZoneName(String zoneName, String errorMessage) {
-        validateNameField(zoneName, DistributionZones.ZONE_DEFINITION_ERR, errorMessage);
-    }
-
-    private static void validateZonePartitions(@Nullable Integer partitions) {
-        validateZoneField(partitions, 1, MAX_PARTITION_COUNT, "Invalid number of partitions");
-    }
-
-    private static void validateZoneReplicas(@Nullable Integer replicas) {
-        validateZoneField(replicas, 1, null, "Invalid number of replicas");
-    }
-
-    private static void validateZoneDataNodesAutoAdjust(@Nullable Integer dataNodesAutoAdjust) {
-        validateZoneField(dataNodesAutoAdjust, 0, null, "Invalid data nodes auto adjust");
-    }
-
-    private static void validateZoneDataNodesAutoAdjustScaleUp(@Nullable Integer dataNodesAutoAdjustScaleUp) {
-        validateZoneField(dataNodesAutoAdjustScaleUp, 0, null, "Invalid data nodes auto adjust scale up");
-    }
+    /**
+     * Validates correctness of incoming value.
+     */
+    public static void validateField(@Nullable Integer value, int min, @Nullable Integer max, String errorPrefix) {
+        if (value == null) {
+            return;
+        }
 
-    private static void validateZoneDataNodesAutoAdjustScaleDown(@Nullable Integer dataNodesAutoAdjustScaleDown) {
-        validateZoneField(dataNodesAutoAdjustScaleDown, 0, null, "Invalid data nodes auto adjust scale down");
+        if (value < min || (max != null && value > max)) {
+            throw new CatalogValidationException(
+                    DistributionZones.ZONE_DEFINITION_ERR,
```

Comment:
let's throw code from catalog's error group ()or just Catalog.VALIDATION_ERR. The same for rest of validation exception